PWGHF/D2H/Tasks/taskDirectedFlowCharmHadrons.cxx

```diff
@@ -0,0 +1,304 @@
+
+    ccdb->setURL(ccdbUrl);
+    ccdb->setCaching(true);
+    ccdb->setLocalObjectValidityChecking();
+  }; // end init
+
+  /// Fill THnSparse
+  /// \param mass is the invariant mass of the candidate
+  /// \param pt is the transverse momentum of the candidate
+  /// \param cent is the centrality of the collision
+  /// \param cosNPhi is the cosine of the n*phi angle
+  /// \param cosDeltaPhi is the cosine of the n*(phi - evtPl) angle
+  /// \param sp is the scalar product
+  /// \param outputMl are the ML scores
+  void fillThn(double& mass,
+               double& pt,
+               double& eta,
+               double& cent,
+               double& uxQxp,
+               double& uyQyp,
+               double& uxQxt,
+               double& uyQyt,
+               std::vector<double>& outputMl,
+               double& sign)
+  {
+    if (storeMl) {
+      registry.fill(HIST("hpuxQxpvscentpteta"), mass, cent, pt, eta, uxQxp, sign, outputMl[0], outputMl[1]);
+      registry.fill(HIST("hpuyQypvscentpteta"), mass, cent, pt, eta, uyQyp, sign, outputMl[0], outputMl[1]);
+      registry.fill(HIST("hpuxQxtvscentpteta"), mass, cent, pt, eta, uxQxt, sign, outputMl[0], outputMl[1]);
+      registry.fill(HIST("hpuyQytvscentpteta"), mass, cent, pt, eta, uyQyt, sign, outputMl[0], outputMl[1]);
+    } else {
+      registry.fill(HIST("hpuxQxpvscentpteta"), mass, cent, pt, eta, uxQxp, sign);
+      registry.fill(HIST("hpuyQypvscentpteta"), mass, cent, pt, eta, uyQyp, sign);
+      registry.fill(HIST("hpuxQxtvscentpteta"), mass, cent, pt, eta, uxQxt, sign);
+      registry.fill(HIST("hpuyQytvscentpteta"), mass, cent, pt, eta, uyQyt, sign);
+    }
+  }
+
+  /// Get the centrality
+  /// \param collision is the collision with the centrality information
+  double getCentrality(CollsWithQvecs::iterator const& collision)
+  {
+    double cent = -999.;
+    switch (centEstimator) {
+      case CentralityEstimator::FV0A:
+        cent = collision.centFV0A();
+        break;
+      case CentralityEstimator::FT0M:
+        cent = collision.centFT0M();
+        break;
+      case CentralityEstimator::FT0A:
+        cent = collision.centFT0A();
+        break;
+      case CentralityEstimator::FT0C:
+        cent = collision.centFT0C();
+        break;
+      default:
+        LOG(warning) << "Centrality estimator not valid. Possible values are V0A, T0M, T0A, T0C. Fallback to V0A";
+        cent = collision.centFV0A();
+        break;
+    }
+    return cent;
+  }
+
+  /// Check if the collision is selected
+  /// \param collision is the collision with the Q vector information
+  /// \param bc is the bunch crossing with timestamp information
+  /// \return true if the collision is selected, false otherwise
+  template <o2::hf_centrality::CentralityEstimator centEstimator>
+  bool isCollSelected(CollsWithQvecs::iterator const& collision,
+                      aod::BCsWithTimestamps const&)
+  {
```

Comment:
This method is not used anywhere.

Reply:
[Reply]: You are correct. I have removed it now from the updated code.